### PR TITLE
ENG-150

### PR DESCRIPTION
**Pull request by Connorbelez**
## ENG-150: Cash Ledger Query System Implementation

This PR implements the remaining cash ledger query functions and middleware to complete the current-state and point-in-time balance query system.

### Key Changes

**New Middleware**
- Added `cashLedgerQuery` and `cashLedgerMutation` middleware with `cash_ledger:view` and `cash_ledger:correct` permissions
- Migrated all existing cash ledger queries from `ledgerQuery` to `cashLedgerQuery` middleware

**New Query Functions**
- `getAccountBalanceRange` - Returns opening/closing balances and entries for a date range with family-aware sign conventions
- `getBorrowerBalance` - Aggregates all BORROWER_RECEIVABLE accounts for a borrower across obligations  
- `getBalancesByFamily` - Groups account balances by family, optionally filtered by mortgage

**Internal Query Variants**
- `internalGetObligationBalance` - Returns obligation balance as number for downstream Convex functions
- `internalGetLenderPayableBalance` - Returns lender payable total as number
- `internalGetMortgageCashState` - Returns family-grouped balances as numbers

**Comprehensive Test Coverage**
- 833 lines of tests covering all new query functions
- Tests for date range partitioning, balance aggregation, auth rejection, and edge cases
- Unit tests for pure balance computation functions
- Integration tests with realistic mortgage/borrower/lender data

### Technical Details

- Uses existing schema indexes (`by_debit_account_and_timestamp`, `by_credit_account_and_timestamp`, etc.)
- Handles entry deduplication for accounts appearing on both debit and credit sides
- Applies correct sign conventions for debit-normal vs credit-normal account families
- Internal variants return `number` instead of `bigint` for Convex function boundary compatibility
- Maintains append-only ledger principles with no state modifications

## Summary by CodeRabbit

* **New Features**
  * Date-range balance queries with opening/closing calculations.
  * Borrower-level receivable aggregation and family-level balance grouping.
  * Permission-gated cash ledger middleware for query/mutation access.
  * Internal variants exposing numeric summaries for downstream consumers.

* **Tests**
  * Comprehensive test suite covering balance partitioning, sign conventions, deduplication, aggregation, and auth/permission checks.

* **Documentation**
  * New spec/docs detailing query behavior, indices, and implementation plan.

### CODE REVIEW

File: convex/fluent.ts [hunk 1]
```diff
@@ -395,6 +395,12 @@ export const ledgerQuery = authedQuery.use(requirePermission("ledger:view"));
 export const ledgerMutation = authedMutation.use(
 	requirePermission("ledger:correct")
 );
+export const cashLedgerQuery = authedQuery.use(
+	requirePermission("cash_ledger:view")
+);
+export const cashLedgerMutation = authedMutation.use(
+	requirePermission("cash_ledger:correct")
+);
```
[the remark this answers is not in the document]
Comment (Connorbelez) [reply convex/fluent.ts]: **Fix:** Updated permission slugs to `cash_ledger:view` / `cash_ledger:correct` (snake_case) to match the implementation in `convex/fluent.ts`. See commit 1357ce0.

File: specs/ENG-150/chunks/manifest.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+## Chunks
+
+| Chunk | Tasks | Status | Description |
+|-------|-------|--------|-------------|
+| chunk-01-remaining-queries | T-001 through T-007 | `pending` | Add cashLedgerQuery middleware, date range query, borrower balance, family aggregation, internal variants, quality gate |
+
+## Notes
+- ENG-148 already implemented 12+ query functions. This chunk fills the remaining gaps from the implementation plan.
+- Single chunk because all tasks are in the same file (queries.ts) with sequential dependencies.
+- Total: 7 tasks, ~200 lines of new code.
```
[the remark this answers is not in the document]
Comment (Connorbelez) [reply specs/ENG-150/chunks/manifest.md]: **Fix:** Updated chunk-01 status from `pending` to `completed` in the manifest. See commit 1357ce0.

File: convex/payments/cashLedger/queries.ts [hunk 3]
```diff
@@ -229,16 +231,194 @@ export const getControlAccounts = ledgerQuery
...
 	.public();
+
+// ── Date Range Queries ───────────────────────────────────────
+
+export const getAccountBalanceRange = cashLedgerQuery
+	.input({
+		accountId: v.id("cash_ledger_accounts"),
+		fromDate: v.string(),
+		toDate: v.string(),
+	})
```
[the remark this answers is not in the document]
Comment (Connorbelez) [reply convex/payments/cashLedger/queries.ts]: **Fix:** Added `BUSINESS_DATE_PATTERN` regex constant (`/^\d{4}-\d{2}-\d{2}$/`) at the top of the file. The `getAccountBalanceRange` handler now validates both `fromDate` and `toDate` against this pattern and throws a clear error if either fails. Also added a `fromDate <= toDate` ordering check. All validation runs before any DB queries. Commit: 41b1289

File: convex/payments/cashLedger/queries.ts [hunk 4]
```diff
@@ -229,16 +231,194 @@ export const getControlAccounts = ledgerQuery
...
+
+		for (const acct of receivables) {
+			const bal = getCashAccountBalance(acct);
+			total += bal;
+			obligations.push({
+				obligationId: acct.obligationId,
+				balance: bal,
+			});
+		}
+
```
[the remark this answers is not in the document]
Comment (Connorbelez) [reply convex/payments/cashLedger/queries.ts]: **Fix:** `getBorrowerBalance` now uses a `Map<Id<"obligations">, bigint>` to group and sum balances by `obligationId`. Accounts without an `obligationId` are skipped. The obligations array type is now `{ obligationId: Id<"obligations">; balance: bigint }` (non-optional). Commit: 41b1289
